### DISCUSSION TITLE
How do we use dark mode and themes?

### DISCUSSION THREAD

**Discussion by viperfx** [tailwindlabs/tailwindcss#4015] [Help]
We recently implemented themes into our application - [see what we did here](https://twitter.com/tharshan_09/status/1377749753757323273)

Now that we are wanting to support dark mode - how should we approach it? Do we have a dark and light variant for each brand theme color? or is there a better way?

**Comment by simonswiss** [comment]
Hmmmm that's an interesting question.

I haven't looked into this with enough consideration, but I'd say that yep, you could define a light/dark color for each CSS color variable.

You could also or redefine the value within a `.dark` scope within your theme scope, if you're using the `class` strategy for dark mode. Doing so, you wouldn't need to use `dark:text-my-color` utilities, the colors would change automatically.

You'd need to carefully choose where you place the `dark` and `my-custom-theme` classes in your document tree, and reflect this in your CSS selector where you redefine the color variables.

If you _do_ want fine-grain control on what colors you apply on each element, I think you need a color set for light and dark for each theme.

**Reply by viperfx** [reply]
Thanks - the dark class should work well for us. Will give that a try.

**Reply by jeanluc-d** [reply]
@simonswiss  
When adding a dark scope into the theme scope, would we do something like this? 
![image](https://user-images.githubusercontent.com/70818495/114926330-374d5100-9dfe-11eb-8fa5-f905c0d4c188.png)
Also does the tailwind.config file need further modification other than adding darkMode: 'class' for the colors to change automatically?
